modelica_builder/transformation.py

```diff
@@ -32,32 +32,54 @@ def build_edits(self, tree, parser):
         return [self.edit(node) for node in selected_nodes]
 
 
-def build_modifications(modifications):
-    """returns a string with modifications formatted properly"""
+def build_modifications(modifications, depth=1, indented=True):
+    """returns a string with modifications formatted properly
+
+    :param modifications: dict
+    :param depth: int, current modification depth, used for determining code indentation
+    :param indent: bool, if true each inserted modification will be indented on new line
+    :return: string
+    """
     mod_strings = []
     for mod_name, mod_value in modifications.items():
         if isinstance(mod_value, dict):
             # treat the modification as a class_modification
-            mod_strings.append(f'{mod_name}({build_modifications(mod_value)})')
+            mod_strings.append(f'{mod_name}({build_modifications(mod_value, depth + 1, indented)})')
         else:
             # treat the modification as a simple assignment
             mod_strings.append(f'{mod_name}={mod_value}')
-    return ', '.join(mod_strings)
+
+    if indented:
+        # depth + 1 because we assume the _root_ thing we're modifying in the
+        # modelica code is already indented once
+        # e.g. a component declaration is usually indented once
+        indent_depth = depth + 1
+        indentation = '\n' + '\t' * indent_depth
+    else:
+        indentation = ''
+
+    return indentation + f',{indentation}'.join(mod_strings)
```

Comment:
This is where the new magic is actually happening. note the assumption. This isn't a very smart formatter, but this should be correct 90% of the time I think.

It's probably worth discussing in the future how opinionated/complicated we should get with formatting as it can quickly complicate our code base and duplicate code for the modelica formatter

Reply:
😬 yeah, do we want to change this into spaces? two or four?

Reply:
https://www.youtube.com/watch?v=SsoOG6ZeyUI -- Richard Hendricks is wrong. 

2 spaces per tab?